smartlearner/tasks/stopping_criteria.py

```diff
@@ -17,3 +19,31 @@ def __init__(self, nb_epochs_max):
     def post_epoch(self, status):
         if status.current_epoch >= self.nb_epochs_max:
             raise TrainingExit(status)
+
+
+class EarlyStopping(Task):
+    def __init__(self, cost, lookahead, callback=None, eps=0.):
+        super(EarlyStopping, self).__init__()
+
+        self.cost = cost
+        self.best_epoch = 0
+        self.best_cost = np.inf
+
+        self.lookahead = lookahead
+        self.callback = callback
+        self.eps = eps
+
+    def init(self, status):
+        self.best_cost = self.cost.view(status)
+
+    def post_epoch(self, status):
+        cost = self.cost.view(status)
+        if cost + self.eps < self.best_cost:
+            self.best_epoch = status.current_epoch
+            self.best_cost = float(cost)
+
+            if self.callback is not None:
```

Comment:
Anything :P. For instance, let's say you want to save the model only when it's getting better, or if you want to print or debug, ... Basically, you passed it any python function that takes a `EarlyStopping` and `Status` objects as arguments.